common/lib/common-definitions/src/logger.ts

```diff
@@ -45,12 +45,166 @@ export interface ITelemetryBaseEvent extends ITelemetryProperties {
     eventName: string;
 }
 
+export type TelemetryErrorCategory = "verbose" | "information" | "warning" | "error" | "critical";
+
+/**
+ * General-use event. These are statically typed events that clients can understand and act on.
+ */
+export interface ITelemetryGenEventBase extends ITelemetryBaseEvent {
+    /*
+     * Flag indicating this is general use event.
+     * Note: It may seem redundant if we end up emitting general use-events separately.
+     * However, it still may be useful if all (dev and general-use) telemetry is routed to a single table.
+     */
+    genUse: true;
+    /*
+     * Subtype (Further description in sublasses).
+     */
+    type: "api" | "service" | "event" | "error";
+    /*
+     * Package this event was emitted from.
+     * It may be useful for consumers to filter out specific packages they are working with.
+     */
+    packageName: string;
+    /*
+     * Class name where this event was emitted from.
+     * It may be useful for consumers to filter out specific classes they are working with.
+     */
+    className: string;
+    /*
+     * Fluid document ID (when applicable). This could be ID, resolved URL etc.
+     */
+    docId?: string;
+    /*
+     * Id of a client who generated this telemetry event (when applicable).
+     */
+    clientId?: string;
```

Comment:
Some observations:
- It would be nice to see a spec clearly articulating user promises. For example, when such event is ever emited, does it mean that class name can't be changed? Code can't move packages?
- Overall, is this an API? With all implications around back-compat, long-term support, etc.?
- Concept like docId does not exists through most of our codebase - we deliberately eliminated it. If it's present on API, where it the meaning of docID is defined? Do 2 different versions of same file (in ODSP) have same docID? There are plenty of scenarios where this term is ambiguous.
- (this is existing problem, but I worry we are making it worse): It's dangerous exposing such concepts as "global" clientId globally, as semantics of this concept (and most importantly - when it changes) different based on layer. For exmaple, Driver, and some parts of Loader layer (DeltaManager) may switch to new clientId while container runtime stays on old one until we catch up / process join op.

Reply:
I'd be curious to learn about prior art. I.e. what are the examples where some system provides stronger guarantees around logging shape, or exposes some perf counters, or otherwise exposes verbose data like that not through direct type-safe API (without type erasure that happens in logging systems).